Include/internal/pycore_freelist.h

```diff
@@ -63,6 +65,31 @@ struct _Py_float_state {
 #endif
 };
 
+#define DICT_MAX_WATCHERS 8
+
+struct _Py_dict_state {
+    /*Global counter used to set ma_version_tag field of dictionary.
+     * It is incremented each time that a dictionary is created and each
+     * time that a dictionary is modified. */
+    uint64_t global_version;
+    uint32_t next_keys_version;
```

Comment:
This is not free-list related state. We still want this to be per-interpreter in the free-threaded builds.